crates/starknet-types-core/src/hash/pedersen.rs

```diff
@@ -17,6 +17,9 @@ impl StarkHash for Pedersen {
 
     /// Computes the Pedersen hash of an array of Felts, as defined
     /// in <https://docs.starknet.io/documentation/architecture_and_concepts/Hashing/hash-functions/#array_hashing.>
+    ///
+    /// Warning: there is room for collision as:
+    /// pedersen_array([value]) and pedersen(pedersen(0, value), 1) will return the same values
```

Comment:
/// Pedersen::hash_array([value]) and Pedersen::hash(Pedersen::hash(0, value), 1) will return the same values